documentation/source/coroutines.rst

```diff
@@ -47,8 +51,25 @@ Coroutines can :keyword:`return` a value, so that they can be used by other coro
         if first == second:
             raise TestFailure("Signal did not change")
 
+Concurrent Execution
+====================
+
+Coroutines can be scheduled for concurrent execution with :func:`~cocotb.fork`, :func:`~cocotb.start`, and :func:`~cocotb.start_soon`.
+
+.. note::
+    :func:`~cocotb.start` and :func:`~cocotb.start_soon` are new scheduling functions.
+    They are future-compatible and are preferred over :func:`~cocotb.fork`.
+
+:func:`~cocotb.fork` schedules and executes the new coroutine immediately,
+returning control to the calling task after the new coroutine finishes or yields control.
+No other pending tasks are run.
```

Comment:
Though it may be beneficial to state that `fork` will be removed without a deprecation notice in the future. Or do you plan on adding a deprecation to `fork` now that the new interfaces exist?

Reply:
We didn't come to a conclusion on whether to remove `fork` or just alias it to `start_soon`.

Reply:
I happen to agree with removal. If that is the decision we need a deprecation before 1.6 is released ideally.

Reply:
I opened an issue about it.